src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/entity/StormStaffLightningListener.java

```diff
@@ -0,0 +1,53 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners.entity;
+
+import javax.annotation.Nonnull;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.LightningStrike;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageByEntityEvent;
+import org.bukkit.event.weather.LightningStrikeEvent;
+
+import io.github.thebusybiscuit.slimefun4.implementation.Slimefun;
+import io.github.thebusybiscuit.slimefun4.implementation.items.magical.staves.StormStaff;
+
+import java.util.UUID;
+
+/**
+ * This {@link Listener} makes sure that if a {@link LightningStrike} spawned by a {@link StormStaff}
+ * causes damage to an {@link org.bukkit.entity.Entity}, the event is transformed so the damage is
+ * attributed to the StormStaff caster.
+ *
+ * @author Sfiguz7
+ *
+ */
+public class StormStaffLightningListener implements Listener {
+
+    public StormStaffLightningListener(@Nonnull Slimefun plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onLightningDamage(EntityDamageByEntityEvent e) {
+        if (e.getDamager() instanceof LightningStrike) {
+            LightningStrike lightningStrike = (LightningStrike) e.getDamager();
+            UUID lightningStrikeUuid = lightningStrike.getUniqueId();
+            if (StormStaff.stormStaffLightnings.containsKey(lightningStrikeUuid)) {
+                e.setCancelled(true);
+                Player caster = Bukkit.getPlayer(StormStaff.stormStaffLightnings.get(lightningStrikeUuid));
```

Comment:
Don't do containsKey and get. `containsKey` calls get. Do the call once and check properly